centralstore/redis_store.go

```diff
@@ -843,24 +849,12 @@ func (t *traceStateProcessor) toNextState(ctx context.Context, conn redis.Conn,
 		"to_state":      changeEvent.next,
 	})
 
-	succeed := make([]string, 0, len(traceIDs))
-	for _, traceID := range traceIDs {
-		err := t.applyStateChange(ctx, conn, changeEvent, traceID)
-		if err != nil {
-			fmt.Println("apply_state_change_error", err)
-			continue
-		}
-		succeed = append(succeed, traceID)
-	}
-	otelutil.AddSpanField(span, "num_of_succeed", len(succeed))
-
-	if len(succeed) == 0 {
-		err := errors.New("failed to apply state change")
-		otelutil.AddSpanField(span, "error", err)
+	result, err := t.applyStateChange(ctx, conn, changeEvent, traceIDs)
+	if err != nil {
```

Comment:
same here -- you don't need to return nil, err, you can just return result, err. If err is non-nil, the value of result is basically undefined so it doesn't matter what you return.